setup.py

```diff
@@ -887,16 +888,24 @@ def configure_extension_build():
 
     # Cross-compile for M1
     if IS_DARWIN:
+        report('-- detected IS_DARWIN')
         macos_target_arch = os.getenv('CMAKE_OSX_ARCHITECTURES', '')
+
         if macos_target_arch in ['arm64', 'x86_64']:
+            report('-- detected macos target architecture ' + macos_target_arch)
             macos_sysroot_path = os.getenv('CMAKE_OSX_SYSROOT')
             if macos_sysroot_path is None:
                 macos_sysroot_path = subprocess.check_output([
                     'xcrun', '--show-sdk-path', '--sdk', 'macosx'
                 ]).decode('utf-8').strip()
             extra_compile_args += ['-arch', macos_target_arch, '-isysroot', macos_sysroot_path]
             extra_link_args += ['-arch', macos_target_arch]
-
+        elif macos_target_arch == '':
+            # We are adding no extra compile and link args. Can this ever be valid?
+            report('-- no target architecture found.')
```

Comment:
This is totally fine, i.e. path taken for the native builds, isn't it?

Reply:
@malfet Thanks tor you comments - can I check what you mean by a native build. Do you mean native as in the ATen native folder devices? If so would this mean `mps` for Apple and we are saying we do not need any extra args for `mps`. I will make the changes assuming this is the case.

Reply:
changed